src/main/java/de/qabel/desktop/DesktopClient.java

```diff
@@ -21,110 +21,150 @@
+            scene = new Scene(new LoginView().getView(), 370, 530);
+            config.addObserver((o, arg) -> {
+                if (arg instanceof Account) {
+                    Scene layoutScene = new Scene(new LayoutView().getView(), 800, 600);
+                    Platform.runLater(() -> primaryStage.setScene(layoutScene));
+                }
+            });
+        } else {
+            scene = new Scene(new LayoutView().getView(), 800, 600);
+        }
+
+        Platform.setImplicitExit(false);
+        primaryStage.setScene(scene);
+        setTrayIcon(primaryStage);
+        primaryStage.setTitle(TITLE);
+        primaryStage.show();
+    }
+
+    private ClientConfiguration initDiContainer(Map<String, Object> customProperties) throws QblInvalidEncryptionKeyException, URISyntaxException {
+        Persistence<String> persistence = new SQLitePersistence("db.sqlite", "qabel".toCharArray(), 65536);
+        customProperties.put("persistence", persistence);
+        customProperties.put("dropUrlGenerator", new DropUrlGenerator("http://localhost:5000"));
+        PersistenceIdentityRepository identityRepository = new PersistenceIdentityRepository(persistence);
+        customProperties.put("identityRepository", identityRepository);
+        PersistenceAccountRepository accountRepository = new PersistenceAccountRepository(persistence);
+        customProperties.put("accountRepository", accountRepository);
+        ClientConfiguration clientConfig = getClientConfiguration(persistence, identityRepository, accountRepository);
+        customProperties.put("clientConfiguration", clientConfig);
+
+        Injector.setConfigurationSource(customProperties::get);
+        Injector.setInstanceSupplier(new RecursiveInjectionInstanceSupplier(customProperties));
+        return clientConfig;
+    }
+
+    private ClientConfiguration getClientConfiguration(Persistence<String> persistence, IdentityRepository identityRepository, AccountRepository accountRepository) {
+        ClientConfigurationRepository repo = new PersistenceClientConfigurationRepository(
+                persistence,
+                new ClientConfigurationFactory(),
+                identityRepository,
+                accountRepository
+        );
+        final ClientConfiguration config = repo.load();
+        config.addObserver((o, arg) -> {
+            repo.save(config);
+        });
+        return config;
+    }
+
+    private void setTrayIcon(Stage primaryStage) throws ClassNotFoundException, UnsupportedLookAndFeelException, InstantiationException, IllegalAccessException {
+
+
+        SystemTray sTray = SystemTray.getSystemTray();
+        primaryStage.setOnCloseRequest(arg0 -> primaryStage.hide());
+
+        final JPopupMenu popup = buildSystemTrayJPopupMenu(primaryStage);
+
+
+
+        URL url = System.class.getResource("/logo.png");
+        Image img = Toolkit.getDefaultToolkit().getImage(url);
+        TrayIcon icon = new TrayIcon(img, "Qabel", null);
```

Comment:
you can omit the 3rd argument (there is a constructor `TrayIcon(Image image, String tooltip)` )